### DISCUSSION TITLE
Can task fail without raising an exception?

### DISCUSSION THREAD

**Discussion by skontar** [django-q2/django-q2#179] [Q&A] [answered]
Hi all! I was thinking if it is possible to make a task fail without raising an exception? I would like to check for a problem and gracefully end the task, but also to communicate to the framework that this task actually failed? Is that possible?

**Comment by GDay** [comment, marked as answer]
That's not possible at the moment. It will only mark a task failed when it raised an exception. You could manually modify the status of the `Task` if you would want to do that to work around this.

**Reply by skontar** [reply]
OK. Thanks for the answer. I guess there is no way to do it from **inside** the task, so I need to create a clever workaround?

**Reply by GDay** [reply]
That's right. When you create task, you get an `id`. You can use that `id` to get the `Task` object in the database. You could use that to mark the task failed manually. 
When a function is running in a worker, that function doesn't have any context of the job itself, so no, I don't think you can do that inside the task. The `Task` object also only gets created after the job has ran.
